dask_ml/model_selection/_incremental.py

```diff
@@ -828,6 +842,7 @@ def __init__(
         self.tol = tol
         self.scores_per_fit = scores_per_fit
         self.max_iter = max_iter
+        self._to_reach = {}
```

Comment:
Mutable variables shouldn't be set in an estimator's `__init__`. Too easy to leak state between successive calls to `.fit`.

AFAICT, this is only used in `self._additional_calls`, and so doesn't need to be a class variable?

I also don't fully understand the reason for this change yet.

Reply:
I added some comment detailing why they're required, and moved them inside the `fit` definition:

* `_steps`: to track the number of times `_additional_calls` has been called. I could probably use the `info` dict passed to `_additional_calls` to get this, but this is far simpler.
* `_pf_calls`: used to record the number of calls for each model. This is required to be a class variable because `info` doesn't have the information for every model. I use this value to make sure that `HyperbandSearchCV` records and passes the correct values to `SuccessiveHalvingSearchCV` (and implemented this test after resolving a bug in December where `max_iter` wasn't passed from `HyperbandSearchCV`).

Reply:
Whoops, I wrote this regarding `_pf_calls` and `_steps` in https://github.com/dask/dask-ml/blob/c6f11af48e17c8b93ae61ffdc22d50434a09c8ff/dask_ml/model_selection/_successive_halving.py#L200 (which also was a mutable state variable declared in `__init__`).

> AFAICT, this is only used in `self._additional_calls`, and so doesn't need to be a class variable?

Correct, it's only used in `self._additional_calls`.

It's a class variable because information needs to be stored about how many more times `partial_fit` needs to be called until the calls specified by `_adapt` are reached. This only matters when `patience` is non-zero (otherwise, `_adapt` calls are scheduled for each model).